### PR TITLE
Deploy function into VNET

### PR DESCRIPTION
**Pull request by akinnane**

### CODE REVIEW

File: data_ingester/terraform/data_ingester/function_rust.tf [hunk 1]
```diff
@@ -47,3 +47,47 @@ resource "azurerm_linux_function_app" "SSPHP_rust" {
...
 }
+
+# Function in VNET
+resource "azurerm_linux_function_app" "SSPHP_rust-vnet" {
+  count                      = var.vnet != null ? 1 : 0
+  name                       = "SSPHP-Metrics-rust-vnet-${local.postfix}"
+  resource_group_name        = azurerm_resource_group.tfstate.name
+  location                   = azurerm_resource_group.tfstate.location
+  tags                       = var.tags
+  storage_account_name       = azurerm_storage_account.tfstate.name
```
Comment (pritchyspritch) [inline data_ingester/terraform/data_ingester/function_rust.tf:58]: Why are you using the terraform state storage as your function app storage? Shouldn't it have it's own?

File: data_ingester/terraform/data_ingester/vnet.tf [hunk 2]
```diff
@@ -0,0 +1,13 @@
+data "azurerm_virtual_network" "vnet" {
+  count               = var.vnet != null ? 1 : 0
+  name                = var.vnet.name
+  resource_group_name = azurerm_resource_group.tfstate.name
```
Comment (pritchyspritch) [inline data_ingester/terraform/data_ingester/vnet.tf:4]: tfstate resource group? Is there not just a generic ssphp rg? Or one specifically for this subnet?

**Review by pritchyspritch — APPROVED** [review]

File: data_ingester/terraform/data_ingester/function_rust.tf [hunk 3]
```diff
@@ -47,3 +47,47 @@ resource "azurerm_linux_function_app" "SSPHP_rust" {
     RUST_LOG                       = "info"
   }
 }
+
+# Function in VNET
+resource "azurerm_linux_function_app" "SSPHP_rust-vnet" {
+  count                      = var.vnet != null ? 1 : 0
```
[the remark this answers is not in the document]
Comment (akinnane) [reply data_ingester/terraform/data_ingester/function_rust.tf:53]: Count is never used. It's a stupid workaround for 'Should I create this resource?'. Happy to be shown a better way.

File: data_ingester/terraform/data_ingester/vnet.tf (hunk 2 above)
Comment (akinnane) [reply data_ingester/terraform/data_ingester/vnet.tf:4]: Agreed to both, and the other comment. Will fix in a future PR as it was a preexisting issue.

**Pull request merged**